dotnetv3/Cognito/Actions/CognitoWrapper.cs

```diff
@@ -0,0 +1,362 @@
+    /// </summary>
+    /// <param name="session"></param>
+    /// <returns></returns>
+    public async Task<string> AssociateSoftwareTokenAsync(string session)
+    {
+        var softwareTokenRequest = new AssociateSoftwareTokenRequest
+        {
+            Session = session,
+        };
+
+        var tokenResponse = await _cognitoService.AssociateSoftwareTokenAsync(softwareTokenRequest);
+        var secretCode = tokenResponse.SecretCode;
+
+        Console.Write("Enter the following token into Google Authenticator: {secretCode}");
+
+        return tokenResponse.Session;
+    }
+
+    // snippet-end:[Cognito.dotnetv3.AssociateSoftwareToken]
+
+    // snippet-start:[Cognito.dotnetv3.AdminInitiateAuth]
+    public async Task<string> AdminInitiateAuthAsync(string clientId, string userPoolId, string userName, string password)
+    {
+        var authParameters = new Dictionary<string, string>();
+        authParameters.Add("USERNAME", userName);
+        authParameters.Add("PASSWORD", password);
+
+        var request = new AdminInitiateAuthRequest
+        {
+            ClientId = clientId,
+            UserPoolId = userPoolId,
+            AuthParameters = authParameters,
+            AuthFlow = AuthFlowType.USER_PASSWORD_AUTH,
+        };
+
+        var response = await _cognitoService.AdminInitiateAuthAsync(request);
+        return response.Session;
+    }
+    // snippet-end:[Cognito.dotnetv3.AdminInitiateAuth]
+
+    // snippet-start:[Cognito.dotnetv3.InitiateAuth]
+    /// <summary>
+    /// Initiate authorization.
+    /// </summary>
+    /// <param name="clientId">The client Id of the application.</param>
+    /// <param name="userName">The name of the user who is authenticating.</param>
+    /// <param name="password">The password for the user who is authenticating.</param>
+    /// <returns>The response from the call to InitiateAuthAsync</returns>
+    public async Task<InitiateAuthResponse> InitiateAuthAsync(string clientId, string userName, string password)
+    {
+        var authParameters = new Dictionary<string, string>();
+        authParameters.Add("USERNAME", userName);
+        authParameters.Add("PASSWORD", password);
+
+        var authRequest = new InitiateAuthRequest
+
+        {
+            ClientId = clientId,
+            AuthParameters = authParameters,
+            AuthFlow = AuthFlowType.USER_PASSWORD_AUTH,
+        };
+
+        var response = await _cognitoService.InitiateAuthAsync(authRequest);
+        Console.WriteLine($"Result Challenge is : {response.ChallengeName}");
+
+        return response;
+    }
+
+    // snippet-end:[Cognito.dotnetv3.InitiateAuth]
+
+    // snippet-start:[Cognito.dotnetv3.ConfirmSignUp]
+    /// <summary>
+    /// Confirm that the user has signed up.
+    /// </summary>
+    /// <param name="clientId">The Id of this application.</param>
+    /// <param name="code">The confirmation code sent to the user.</param>
+    /// <param name="userName">The user name.</param>
```

Comment:
user name --> username 
(here and throughout the file)

Reply:
fixed throughout the solution